diff --git a/src/overlay/graphs.c b/src/overlay/graphs.c
index 21f2251..fbc7304 100644
--- a/src/overlay/graphs.c
+++ b/src/overlay/graphs.c
@@ -197,7 +197,8 @@ static int graphs_render(void * udata, bool interactive,
       title = _title;
     }
 
+    igPushID_Ptr(graph);
     igPlotLines_FloatPtr(
-        (void *)graph,
+        "##timingGraph",
         (float *)ringbuffer_getValues(graph->buffer),
         ringbuffer_getLength(graph->buffer),
@@ -207,7 +208,8 @@ static int graphs_render(void * udata, bool interactive,
         graph->min,
         graph->max,
         (ImVec2){ winSize.x, height },
-        sizeof(float));
+        sizeof(float));
+    igPopID();
   };
